web/src/main/java/cz/muni/fi/pa165/team/match/MatchController.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @RequestMapping(value = "/matches", method = RequestMethod.GET)
+    public ModelAndView viewListPlayed()
+    {
+        return new ModelAndView("team/match/played")
+            .addObject("matchResults", teamMatchFacade.getPlayedMatchesList())
+            .addObject("formatter", DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+    }
+
+    @RequestMapping(value = "/matches/upcomming", method = RequestMethod.GET)
+    public ModelAndView viewListUpcomming()
+    {
+        return new ModelAndView("team/match/upcomming")
+            .addObject("plannedMatches", teamMatchRepository.findAllPlannedMatches())
+            .addObject("formatter", DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+    }
+
+    @RequestMapping(value = "/match/create", method = RequestMethod.GET)
+    public ModelAndView viewNewMatch()
+    {
+        return new ModelAndView("team/match/create")
+            .addObject("teams", teamRepository.findAll())
+            .addObject("matchRequest", new MatchRequest());
+    }
+
+    @RequestMapping(value = "/match/create", method = RequestMethod.POST)
+    public ModelAndView submitNewMatch(
+        @ModelAttribute("matchRequest") @Validated MatchRequest matchRequest,
+        BindingResult result
+    ) throws Exception
+    {
+        if (result.hasErrors()) {
+            return new ModelAndView("team/match/create")
+                .addObject("teams", teamRepository.findAll())
+                .addObject("matchRequest", matchRequest);
+        }
+
+        LocalDateTime startTime = getMVCLocalDateTime(matchRequest.getStartTime());
+        LocalDateTime endTime = getMVCLocalDateTime(matchRequest.getEndTime());
+
+        TeamMatch teamMatch = teamMatchFacade.createMatch(
+            UUID.fromString(matchRequest.getHomeTeamId()),
+            UUID.fromString(matchRequest.getAwayTeamId()),
+            startTime,
+            endTime
+        );
+
+        return new RedirectResponse("/match/" + teamMatch.getId().toString());
+    }
+
+    private LocalDateTime getMVCLocalDateTime(String formInputDate)
+    {
+        if (formInputDate == null || formInputDate.isEmpty()){
+            return null;
+        }
+
+        String[] splittedFormDate = formInputDate.split("T");
+        String dateToBeParsed = splittedFormDate[0] + " " + splittedFormDate[1];
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+        return LocalDateTime.parse(dateToBeParsed, formatter);
+    }
+
+    /**
+     * Validates match form data.
+     */
+    class MatchCreationValidator implements Validator
+    {
+
+        @Override
+        public boolean supports(Class type)
+        {
+            return MatchRequest.class.isAssignableFrom(type);
+        }
+
+        @Override
+        public void validate(Object target, Errors errors)
+        {
+            MatchRequest request = (MatchRequest) target;
+
+            if (request.getHomeTeamId() == null) {
```

Comment:
Kde presne by se ta anotace @NotNull mela pouzit? zkousel jsem to pouzit nad atributy request objektu (protoze ne vsechny atributy nemuzou byt null), ale to rozhodne nefunguje...